great_expectations/core/usage_statistics/anonymizers/batch_request_anonymizer.py

```diff
@@ -0,0 +1,109 @@
+import copy
+from typing import Any, Dict, List, Optional, Set, Union
+
+from great_expectations.core.batch import (
+    BatchRequest,
+    RuntimeBatchRequest,
+    get_batch_request_from_acceptable_arguments,
+)
+from great_expectations.core.usage_statistics.anonymizers.anonymizer import Anonymizer
+from great_expectations.util import deep_filter_properties_dict
+
+from great_expectations.core.batch import (  # isort:skip
+    BATCH_REQUEST_REQUIRED_TOP_LEVEL_KEYS,
+    BATCH_REQUEST_OPTIONAL_TOP_LEVEL_KEYS,
+    DATA_CONNECTOR_QUERY_KEYS,
+    RUNTIME_PARAMETERS_KEYS,
+    BATCH_SPEC_PASSTHROUGH_KEYS,
+    BATCH_REQUEST_FLATTENED_KEYS,
+)
+
+
+class BatchRequestAnonymizer(Anonymizer):
+    def __init__(self, salt=None):
+        super().__init__(salt=salt)
+
+        self._batch_request_optional_top_level_keys = []
+        self._data_connector_query_keys = []
+        self._runtime_parameters_keys = []
+        self._batch_spec_passthrough_keys = []
+
+    def anonymize_batch_request(self, *args, **kwargs) -> Dict[str, List[str]]:
+        batch_request: Union[
+            BatchRequest, RuntimeBatchRequest
+        ] = get_batch_request_from_acceptable_arguments(*args, **kwargs)
+        batch_request_dict: dict = batch_request.to_json_dict()
+        anonymized_batch_request_dict: Optional[
+            Union[Any, dict]
+        ] = self._anonymize_batch_request_properties(source=batch_request_dict)
+        deep_filter_properties_dict(
+            properties=anonymized_batch_request_dict,
+            delete_fields=BATCH_REQUEST_REQUIRED_TOP_LEVEL_KEYS,
+            clean_falsy=True,
+            inplace=True,
+        )
+        anonymized_batch_request_keys_dict: Dict[str, List[str]] = {
+            "batch_request_optional_top_level_keys": self._batch_request_optional_top_level_keys,
+            "data_connector_query_keys": self._data_connector_query_keys,
+            "runtime_parameters_keys": self._runtime_parameters_keys,
+            "batch_spec_passthrough_keys": self._batch_spec_passthrough_keys,
+        }
+        self._build_anonymized_batch_request(source=anonymized_batch_request_dict)
+        deep_filter_properties_dict(
+            properties=anonymized_batch_request_keys_dict,
+            clean_falsy=True,
+            inplace=True,
+        )
+        return anonymized_batch_request_keys_dict
+
+    def _anonymize_batch_request_properties(
+        self, source: Optional[Any] = None
+    ) -> Optional[Union[Any, dict]]:
```

Comment:
Could this instead be Optional[dict]? Any time I see this I think of the poor caller, who has to say "if None .... if dict ... else (Any) ..."

Reply:
@donaldheppner I did this pattern for clarity -- to show that we can have either a dictionary or a single item of a non-dictionary type.  Thanks.

Reply:
I understand the type hint, my question is if we want to support returning a single non-dictionary item (on considering this more deeply, I think we do). At the very least, per the implementation, could we make the type hint for the return type Optional[Union[str, dict]]?

Reply:
@donaldheppner Some times the values are not a string -- that is what the recursive procedure handles when we simply return the source (which could be any value -- for example, `limit` is an integer).

Reply:
@donaldheppner and I just discussed this, and @donaldheppner is absolutely correct.  I will apply the fix in the next commit.  Thank you!